cluster-autoscaler/main.go

```diff
@@ -455,7 +430,7 @@ func buildAutoscaler(debuggingSnapshotter debuggingsnapshot.DebuggingSnapshotter
 	}
 	informerFactory := informers.NewSharedInformerFactoryWithOptions(kubeClient, 0, informers.WithTransform(trim))
 
-	eventsKubeClient := createKubeClient(getKubeConfig())
+	eventsKubeClient := createKubeClient(kube_util.GetKubeConfig(autoscalingOptions))
```

Comment:
Is `KubeConfig` abstraction needed at all in main? Can the entire client creation be moved to a dedicated package?

Reply:
Yes. releated fields and `getKubeConfig`、`createKubeClient` now are only used in `main.go`. I do this refactor because I want to reuse these functions in cloudprovider.